File: src/analyzer/predicate-container-type.ts
export const PREDICATE_CONTAINER_TYPE = {
  IfStatement: "IfStatement",
  SwitchStatement: "SwitchStatement",
  WhileStatement: "WhileStatement",
  DoWhileStatement: "DoWhileStatement",
  ForStatement: "ForStatement",
  ForOfStatement: "ForOfStatement",
  ForInStatement: "ForInStatement",
} as const;
export type PredicateContainerType =
  (typeof PREDICATE_CONTAINER_TYPE)[keyof typeof PREDICATE_CONTAINER_TYPE];

File: crates/unsnarl-ir/src/predicate_container_type.rs
//! Predicate container categorization. Ports
//! `ts/src/analyzer/predicate-container-type.ts`.

use serde::Serialize;

#[derive(Serialize)]
pub enum PredicateContainerType {
    IfStatement,
    SwitchStatement,
    WhileStatement,
    DoWhileStatement,
    ForStatement,
    ForOfStatement,
    ForInStatement,
}
